### DISCUSSION TITLE
Hide copilot suggestions when inserting snippets

### DISCUSSION THREAD

**Discussion by comiluv** [L3MON4D3/LuaSnip#969] [Q&A]
I'm trying to use LuaSnip with https://github.com/zbirenbaum/copilot.lua
What I want is to hide copilot suggestions when the cursor is focused on LuaSnip snippets
So that I can hide copilot suggestions when using tab key to jump forth snippet locations
My current implementation is as below. Using lazy.nvim,
```lua
config = function()
	require'luasnip'.setup({})
	vim.api.nvim_create_autocmd("User", {
		pattern = "LuasnipInsertNodeEnter",
		callback = function()
			vim.b.copilot_suggestion_hidden = true
		end,
	})
	vim.api.nvim_create_autocmd("User", {
		pattern = "LuasnipInsertNodeLeave",
		callback = function()
			vim.b.copilot_suggestion_hidden = false
		end,
	})
end
```
Is there any better way than this?

**Comment by L3MON4D3** [comment]
Seems okay, it obviously depends on your workflow.
What would be your optimal criterion for dis/enabling copilot?
I can see this working pretty well if `region_check_events` is set to some common event (since then the nodes are left after they are no longer edited)

**Reply by comiluv** [reply]
Thanks for your answer.
The down side of the above set up code is that if the user presses \<Esc> or \<C-c> to exit from the insert mode while the cursor is located on a snippet location, LuasnipInsertNodeLeave is not triggered and the variable is not set to false.

I suppose I could make another autocmd on "InsertLeave" like below to capture such user input for more consistent experience but I feel it's kind of hacky and would like more elegant approach for this.
```lua
vim.api.nvim_create_autocmd("InsertLeave", {
  group = luasnip_group,
  callback = function()
    vim.b.copilot_suggestion_hidden = false
  end,
})
```
To note is that I have also set up nvim-cmp for completions like below to toggle copilot when the completions menu is up, which works well for me.
```lua
cmp.event:on("menu_opened", function()
  vim.b.copilot_suggestion_hidden = true
end)
cmp.event:on("menu_closed", function()
  local luasnip = require("luasnip")
  if not (luasnip.jumpable(1) or luasnip.jumpable(-1)) then
    vim.b.copilot_suggestion_hidden = false
  end
end)
```
